concrete/src/Page/Collection/Version/Version.php

```diff
@@ -661,55 +800,143 @@ public function delete()
             $fa->delete();
         }
 
-        $category = $app->make('Concrete\Core\Attribute\Category\PageCategory');
+        $category = $this->getObjectAttributeCategory();
         $attributes = $category->getAttributeValues($this);
         foreach ($attributes as $attribute) {
             $category->deleteValue($attribute);
         }
 
-        $db->executeQuery('delete from CollectionVersionBlockStyles where cID = ? and cvID = ?', array(
-            $cID,
-            $cvID,
-        ));
-        $db->executeQuery('delete from CollectionVersionThemeCustomStyles where cID = ? and cvID = ?', array(
-            $cID,
-            $cvID,
-        ));
-        $db->executeQuery('delete from CollectionVersionRelatedEdits where cID = ? and cvID = ?', array(
-            $cID,
-            $cvID,
-        ));
-        $db->executeQuery('delete from CollectionVersionAreaStyles where cID = ? and cvID = ?', array(
-            $cID,
-            $cvID,
-        ));
-
-        $db->executeQuery('delete from PageTypeComposerOutputBlocks where cID = ? and cvID = ?', array(
-            $cID,
-            $cvID,
-        ));
-
-        $q = "delete from CollectionVersions where cID = '{$cID}' and cvID='{$cvID}'";
-        $db->executeQuery($q);
+        $db->executeQuery(
+            'delete from CollectionVersionBlockStyles where cID = ? and cvID = ?',
+            [$cID, $cvID]
+        );
+        $db->executeQuery(
+            'delete from CollectionVersionThemeCustomStyles where cID = ? and cvID = ?',
+            [$cID, $cvID]
+        );
+        $db->executeQuery(
+            'delete from CollectionVersionRelatedEdits where cID = ? and cvID = ?',
+            [$cID, $cvID]
+        );
+        $db->executeQuery(
+            'delete from CollectionVersionAreaStyles where cID = ? and cvID = ?',
+            [$cID, $cvID]
+        );
+        $db->executeQuery(
+            'delete from PageTypeComposerOutputBlocks where cID = ? and cvID = ?',
+            [$cID, $cvID]
+        );
+
+        $db->executeQuery(
+            'delete from CollectionVersions where cID = ? and cvID = ?',
+            [$cID, $cvID]
+        );
         $this->refreshCache();
     }
 
-    private function clearPublishStartDate()
+    /**
+     * {@inheritdoc}
+     *
+     * @see \Concrete\Core\Permission\ObjectInterface::getPermissionObjectIdentifier()
+     */
+    public function getPermissionObjectIdentifier()
     {
-        $app = Facade::getFacadeApplication();
-        $db = $app->make('database')->connection();
-        $q = "update CollectionVersions set cvPublishDate = NULL where cID = ?";
+        return $this->getCollectionID() . ':' . $this->getVersionID();
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @see \Concrete\Core\Permission\ObjectInterface::getPermissionResponseClassName()
+     */
+    public function getPermissionResponseClassName()
+    {
+        return '\\Concrete\\Core\\Permission\\Response\\CollectionVersionResponse';
```

Comment:
@mlocati I think this can be just `\Concrete\Core\Permission\Response\CollectionVersionResponse::class`, don't you think?

Reply:
Me too I like the `::class` approach, so that my IDE lets me jump to the class definition: PR updated.